include/odp/api/spec/packet_io.h

```diff
@@ -1255,6 +1256,77 @@ typedef struct odp_pktio_info_t {
  */
 int odp_pktio_info(odp_pktio_t pktio, odp_pktio_info_t *info);
 
+/**
+ * Packet IO link speed in Mbps
+ */
+typedef enum odp_pktio_link_speed_t {
+	ODP_PKTIO_LINK_SPEED_UNKNOWN = 0,
+	ODP_PKTIO_LINK_SPEED_10M    = 10,
+	ODP_PKTIO_LINK_SPEED_100M   = 100,
+	ODP_PKTIO_LINK_SPEED_1G     = 1000,
+	ODP_PKTIO_LINK_SPEED_2_5G   = 2500,
+	ODP_PKTIO_LINK_SPEED_5G     = 5000,
+	ODP_PKTIO_LINK_SPEED_10G    = 10000,
+	ODP_PKTIO_LINK_SPEED_20G    = 20000,
+	ODP_PKTIO_LINK_SPEED_25G    = 25000,
+	ODP_PKTIO_LINK_SPEED_40G    = 40000,
+	ODP_PKTIO_LINK_SPEED_50G    = 50000,
+	ODP_PKTIO_LINK_SPEED_56G    = 56000,
+	ODP_PKTIO_LINK_SPEED_100G   = 100000,
+	ODP_PKTIO_LINK_SPEED_200G   = 200000,
+	ODP_PKTIO_LINK_SPEED_400G   = 400000
+} odp_pktio_link_speed_t;
+
+/** Autonegotiation mode */
+typedef enum odp_pktio_link_autoneg_t {
+	/** Autonegotiation disabled */
+	ODP_PKTIO_LINK_FIXED = 0,
+	/** Autonegotiation enabled */
+	ODP_PKTIO_LINK_AUTONEG  = 1
+} odp_pktio_link_autoneg_t;
+
+/** Duplex mode */
+typedef enum odp_pktio_link_duplex_t {
+	ODP_PKTIO_LINK_DUPLEX_UNKNOWN = 0,
+	ODP_PKTIO_LINK_DUPLEX_HALF = 1,
+	ODP_PKTIO_LINK_DUPLEX_FULL = 2
+} odp_pktio_link_duplex_t;
+
+/** Link status */
+typedef enum odp_pktio_link_status_t {
+	ODP_PKTIO_LINK_DOWN = 0,
+	ODP_PKTIO_LINK_UP = 1
+} odp_pktio_link_status_t;
+
+/**
+ * Packet IO link information
+ */
+typedef struct odp_pktio_link_info_t {
+	/** Link autonegotiation */
+	odp_pktio_link_autoneg_t autoneg;
+	/** Duplex mode */
+	odp_pktio_link_duplex_t duplex;
+	/** Link speed in Mbps */
+	odp_pktio_link_speed_t speed;
+	/** Link status */
+	odp_pktio_link_status_t status;
+} odp_pktio_link_info_t;
```

Comment:
Link flow control status could be also added here (pause frame on vs off)

Reply:
Not sure, drivers can reliability get the exact "live" status of the pause frame.  i.e by the time odp_pktio_link_status() sync call get completed, it may not report the exact status of pause frame as it will keep changing based on the peer load, unlike link up/down event where up/down takes time to transition from one state or another state.

Reply:
I guess Petri was thinking more about the configured value (enabled/disabled) and not the live status. The current plan is to add further configuration APIs in the near future. Perhaps it would make sense to wait with the pause status until we are clear how the matching configure path is implemented.

Reply:
OK. If it is not the live status then fine.

Reply:
Another thing to add could be link media: e.g. twisted pair/copper, fiber, etc. Enumeration may be tricky. Copper vs fiber is likely the most common and usable values. E.g. values could be: unknown, copper, fiber, wireless, other.